### PR TITLE
Add query planning CPU time stat

### LINKED ISSUE #2579: Expose planningCpu time in QueryStats
We could expose total CPU planning time in `QueryStats`. This would simplify debugging planning issues and latency.

### PR DESCRIPTION
**Pull request by weijiii**
## Description
This PR aims to close #2579 

I think the thread in which planning begins does not need to be tracked when writing to `QueryStateTimer#planningCpuTime`. For DDL queries the whole state transitioning happens on the same thread. For normal queries `beginPlanning` and `beginStarting` are supposed to run on the same thread in a linear fashion, hence the first attempt to set `planningCpuTime` should yield correct result.

The following image shows the change on the UI. I suppose this should also be in `QueryCompletedEvent` so I will work on that and update this PR.

Screenshot 2023-04-01 at 3 35 18 AM

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by weijiii** [comment]
Seems a system connector test is failing. Will fix.

**Comment by weijiii** [comment]
Added item in `trino-spi/pom.xml` because number of parameters changed for `QueryStatistics`

**Comment by weijiii** [comment]
@sopel39 Hi, can you take a look? Thanks.

**Review by sopel39 — APPROVED** [review]
lgtm % I think you should extract mysql event listener changes to separate (follow-up) PR

File: plugin/trino-mysql-event-listener/src/main/java/io/trino/plugin/eventlistener/mysql/MysqlEventListener.java [hunk 1]
```diff
@@ -131,6 +131,7 @@ public void queryCompleted(QueryCompletedEvent event)
                 stats.getResourceWaitingTime().map(Duration::toMillis).orElse(0L),
                 stats.getAnalysisTime().map(Duration::toMillis).orElse(0L),
                 stats.getPlanningTime().map(Duration::toMillis).orElse(0L),
+                stats.getPlanningCpuTime().map(Duration::toMillis).orElse(0L),
```
[the remark this answers is not in the document]
Comment (arhimondr) [reply plugin/trino-mysql-event-listener/src/main/java/io/trino/plugin/eventlistener/mysql/MysqlEventListener.java:134]: This change is braking. Though the MySQL listener is not designed to be used in production as of today and is built for testing purposes only. We don't have to maintain strong compatibility.

**Review by arhimondr — APPROVED** [review]

File: plugin/trino-mysql-event-listener/src/main/java/io/trino/plugin/eventlistener/mysql/MysqlEventListener.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sopel39) [reply plugin/trino-mysql-event-listener/src/main/java/io/trino/plugin/eventlistener/mysql/MysqlEventListener.java:134]: Thanks. In that case I thin MySQL listener should be out of scope of this PR

[the remark this answers is not in the document]
Comment (weijiii) [reply plugin/trino-mysql-event-listener/src/main/java/io/trino/plugin/eventlistener/mysql/MysqlEventListener.java:134]: @sopel39 @arhimondr 
Thanks for the review. I'd like to finish this PR. I will
* Rebase w.r.t the latest commit
* Extract the changes on `trino-mysql-event-listener` to a separate PR

**Comment by weijiii** [comment]
I've rebased and extracted the changes on `trino-mysql-event-listener` to https://github.com/trinodb/trino/pull/16841
The current test failures are due to `trino-mysql-event-listener` build. Please take a look. Thanks! 
cc: @sopel39 @arhimondr

**Comment by sopel39** [comment]
> The failures are caused by the missing parts for trino-mysql-event-listener

Could you make the build pass?

**Comment by weijiii** [comment]
> > For all CI tasks to pass I will need to include the changes on mysql event listener.
> 
> I'm not sure why. You can have extra field in `QueryStatistics`, but not necessarily mysql changes as part of this PR. The UI change on it's own is very useful

Thanks let me try that.

**Comment by weijiii** [comment]
@sopel39 Rebased and excluded `trino-mysql-event-listener` DAO

File: core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java [hunk 2]
```diff
@@ -683,6 +683,7 @@ private QueryStats getQueryStats(Optional<StageInfo> rootStage, List<StageInfo>
                 queryStateTimer.getAnalysisTime(),
                 queryStateTimer.getPlanningTime(),
                 queryStateTimer.getFinishingTime(),
+                queryStateTimer.getPlanningCpuTime(),
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java:686]: could you move it immediately below `getPlanningTime()` (here and in other places)?

**Review by sopel39 — APPROVED** [review]
lgtm % comment

File: core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java (hunk 2 above)
Comment (weijiii) [reply core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java:686]: Sure

**Comment by sopel39** [comment]
merged, thanks!

**Pull request merged**